### ISSUE TITLE
[Bug]: wrong commit : the message is not about the diff 

### ISSUE THREAD

**Issue by iauhc** [di-sukharev/opencommit#252]
### Opencommit Version

v2.4.2

### Node Version

v20.1.0

### NPM Version

v9.8.0

### What OS are you seeing the problem on?

Other Linux Distro

### What happened?

The commit seems bugged.  
.... (about 20 other files but not server.ts at all)
  roa/views/formation/liste_formation_programme.php
  roa/views/formation/voir_formation_programme.php
  autocron.php
│

◇  📝 Commit message generated
│
└  Commit message:
——————————————————
fix(server.ts): change the case of the port variable from lowercase to uppercase (PORT)
feature(server.ts): add support for the process.env.PORT environment variable

### Expected Behavior

Right commit message, at least about the filed changed

### Current Behavior

Broken commit message

**Comment by malpou** [comment]
Seems like something weird is happening causing the model to use the translation files example as the response messages for the commit:
https://github.com/di-sukharev/opencommit/blob/master/src/i18n/en.json

**Comment by iauhc** [comment]
Hmm, I use it in French, maybe this can explain. Will switch in en

**Comment by malpou** [comment]
Also, there is now a version 3 out, I see you're on 2.4.2 so upgrading might give another result.

**Comment by di-sukharev** [comment]
@iauhc please close the issue if solved

**Comment by Paillat-dev** [comment]
Still there in 3.0.4.

**Comment by Sped0n** [comment]
same here, It could be because there is too much code in the diff.

<img width="1512" alt="image" src="https://github.com/di-sukharev/opencommit/assets/70063177/0dc6039a-9945-4820-97b2-add8c2c1e8dc">

**Comment by Paillat-dev** [comment]
Stale but not this is an actual prompting issue wich breaks half of the commit messages

**Comment by di-sukharev** [comment]
reopening, will take a look at this

**Comment by tomasvts** [comment]
I just experience this in a small commit
```
◇  📝 Commit message generated
│
└  Generated commit message:
——————————————————
 Changed port variable name in server.ts and added support for configurable port.
```

The problem is in the prompts.ts file https://github.com/di-sukharev/opencommit/blob/master/src/prompts.ts#L37 which is adding an example result prompt to the engine, and sometimes, the engine might grab it as an actual git diff.

**Comment by Paillat-dev** [comment]
Yup, also, idk what models this is using, but if it is still using complete model it should imo switch to chat models. Also, the prompts need to be changed since the models where updated, they don't work the same anymore

**Comment by Paillat-dev** [comment]
I did notice that too, however, I also noticed that depending on the model, it dosen't follow well some instructions, like adding a description, and also, the emojis used are always wrong as per [gitmoji](https://gitmoji.dev/).

**Comment by di-sukharev** [comment]
heads down, the prompt is weak, it can be 10x better, it happens to fail on large diffs, we should improve `splitDiff()` function and definitely rework the prompt and make it also configurable, so everybody can play with it locally, any ideas?

**Comment by Paillat-dev** [comment]
Dropping some ideas here, it could be nice to be able to set custom settings per repository, including prompt context that a user can add (ex. Your repo is about a collection of separate bash scripts, it could be useful for the model to know that they aren't linked together). Maybe some config folder like `.oco` or something. Also, standardizing the prompt formats would allow to easily add new ones, maybe even locally (ex. prompts need to contain `<<DIFF>>`, `<<REPO_NAME>>`, `<<USER_CONTEXT>>` which willl then be replaced programmatically. I am not familiar with ts tough so I did not dig too much in this codebase.

**Comment by di-sukharev** [comment]
@Paillat-dev or it can be learning on the repo with every commit and make better commits bit by bit 💣  not sure it will help though..

as for the different commit message — its not a bug with library, gpt is simply confused in a large git diff. If the there are too many changes — we can do multiple commits instead of just one, who wants to do a PR, @Paillat-dev would you like to contribute to the repo?

as a temporary workaround, commit in batches, e.g. by folders or commit large files separately

**Comment by Paillat-dev** [comment]
As I said I almost never touched Typescript nor JavaScript in my entire life so I don't think I can do much other than opening issues 😅

**Comment by digitalstudium** [comment]
Same shit.
```
◇  2 staged files:
  03_network/extra_kube_yaml/ldap.yaml
  08_sonarqube/values/sq.yaml
│
◇  📝 Commit message generated
│
└  Generated commit message:
——————————————————
**1. server.ts: fix port variable case, add support for configurable port**
2. Create LDAP Service and Endpoints configuration file
3. Update SonarQube values with new properties and security configurations
```
It would be wonderful to remove this first message... Tried to investigate source code of opencommit but failed, because I'm not a JS dev

**Comment by digitalstudium** [comment]
And even when I change a single file, same thing happens. 

```
$ opencommit
┌  open-commit
│
◇  1 staged files:
  08_sonarqube/values/sq.yaml
│
◇  📝 Commit message generated
│
└  Generated commit message:
——————————————————
 fix(server.ts): change port variable case (fix semantics)
feat(server.ts): add support for configurable port (run app on a desired port)

modify(sq.yaml): enable persistence and set up SonarQube with LDAP authentication.

```

**Comment by toby1991** [comment]
I'd like to say, `server.ts` is a mess..

**Comment by di-sukharev** [comment]
there is no `server.ts`, it's a few shot that shows an LLM how to structure commits, this can happen on small commits, but worth investigating and fixing, please dont hesitate to open a pr

**Comment by toby1991** [comment]
> there is no `server.ts`, it's a few shot that shows an LLM how to structure commits, this can happen on small commits, but worth investigating and fixing, please dont hesitate to open a pr

I think the way to resolve this, may change the prompt, to explicitly point out "if no `server.ts` changes, only show the diffs", and make the translation example constantly.

I think this will change the main behavior of commit generating, that I dont think I have the ability to make it working properly through my current prompt skill, so I just, "point out" @di-sukharev

**Comment by di-sukharev** [comment]
This is still a valid prompt-grounding issue and is the canonical tracker for the same sample-diff hallucination reported in #302. The current prompt still includes a concrete `server.ts` few-shot example, which smaller/local models can copy instead of using the real diff.

Suggested implementation:
1. Clearly delimit and label examples, then explicitly state that example filenames/content must never appear unless present in the actual diff.
2. Consider replacing the concrete example with a more abstract schema for weaker local models.
3. Add prompt snapshot tests and a deterministic fake-engine regression that verifies the real diff is the final, clearly delimited user input.

Feel free to open a pull request against `master` and reference this issue.
